### ISSUE TITLE
Update devise_saml_authenticatable gem

### ISSUE THREAD

**Issue by mattbaumann1** [NUARIG/competitions#519]
Update devise_saml_authenticatable to the newest version of master branch. 

The newest version adds the ability to set the attribute-map in the /config/initializers/devise.rb. This can be combined with a call to the competitions_config.yml and then we can eliminate the /config/attribute-map.yml file. 

https://github.com/apokalipto/devise_saml_authenticatable/pull/162

- [x] upgrade gem and remove call to github
- [x] runs specs
- [x] move the attribute map

**Comment by mattbaumann1** [comment]
We are waiting on making these changes until the NU instance is configured, so this is blocked for now. #438

**Comment by mattbaumann1** [comment]
Corresponded with maintainer and they are going to change readme to reflect that you need to pull gem from the GitHub repo.

**Comment by mattbaumann1** [comment]
There are a series of specs that broke unassociated with the changes in this ticket. They will be dealt with here. It was an issue with Capybara. 

https://github.com/teamcapybara/capybara#modals

**Comment by mattbaumann1** [comment]
There are intermittent failures in the following file. These also fail intermittently on develop, so a new ticket ( #540  ) will be opened for this. 
rspec ./spec/system/grants/criteria_spec.rb
